packages/core/src/parameter/types.ts

```diff
@@ -72,10 +72,11 @@ interface TypedCommandFlagParameters_<FLAGS extends BaseFlags, CONTEXT extends C
     readonly aliases?: Aliases<keyof FLAGS & string>;
 }
 
-export type TypedCommandFlagParameters<FLAGS extends BaseFlags, CONTEXT extends CommandContext> =
-    Record<string, never> extends FLAGS
-        ? Partial<TypedCommandFlagParameters_<FLAGS, CONTEXT>>
-        : TypedCommandFlagParameters_<FLAGS, CONTEXT>;
+export type TypedCommandFlagParameters<FLAGS extends BaseFlags, CONTEXT extends CommandContext> = [
+    keyof FLAGS,
+] extends [never]
+    ? Partial<TypedCommandFlagParameters_<FLAGS, CONTEXT>>
+    : TypedCommandFlagParameters_<FLAGS, CONTEXT>;
```

Comment:
What is the motivation for changing this conditional?

Reply:
It goes hand in hand with the default type parameter change [here](https://github.com/bloomberg/stricli/pull/84#discussion_r2064871733)